### PR TITLE
300+ range treated as errors. Do not shorten error body so we can mor…

### PR DESCRIPTION
**Pull request by twhiston**
…e easily diagnose issues

### CODE REVIEW

File: lib/netbox_client_ruby/communication.rb [hunk 1]
```diff
@@ -45,8 +45,8 @@ def raise_on_http_error(response)
       body = response.body
 
       case status
-      when 200..399 then
-      when 400..499 then
+      when 200..299 then
+      when 300..499 then
```
Comment (cimnine) [inline lib/netbox_client_ruby/communication.rb]: Why's this better? The original logic was that `200..399` are all considered success codes, whereas `400..499` are definitely failure codes.

Comment (twhiston) [reply lib/netbox_client_ruby/communication.rb]: See https://redmine.nine.ch/issues/38280 for the issue.
We do not want to follow 302 redirects and as such we will treat them as errors.

Comment (cimnine) [reply lib/netbox_client_ruby/communication.rb]: Of course. I miss-interpreted the ruby `case ... when` syntax. (It does not 'fall through' if nothing is in a `when` block.)

Comment (twhiston) [reply lib/netbox_client_ruby/communication.rb]: I also assumed it would fall through when I first looked at it

Comment (twhiston) [reply lib/netbox_client_ruby/communication.rb]: Apparently I can't alter that commit message as it refuses to push if i try to amend the last commit.

**Pull request merged**